spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -1021,7 +1022,32 @@ struct ProductionVisitor
                 auto pstate = pb->state();
                 pstate.literal_mode = mode;
                 pushState(std::move(pstate));
+
+                // Since `advance` can trigger recoverable errors when
+                // hitting a gap, bracket the call to it in a `try`/`catch`
+                // block if we are in search mode and attempt to recover.
+                if ( mode == LiteralMode::Search ) {
+                    auto [body, try_] = builder()->addTry();
+
+                    pushBuilder(try_.addCatch(builder::parameter(ID("e"), builder::typeByID("hilti::MissingData"))),
```

Comment:
This is being popped again later? (Can't quite tell)

Reply:
This doesn't need to be popped since the overload here is the form which directly takes a callback and handles push/pop for the user.